src/AWS.Deploy.Orchestration/DeploymentBundleHandler.cs

```diff
@@ -103,6 +104,13 @@ public async Task<string> CreateDotnetPublishZip(Recommendation recommendation)
             _interactiveService.LogInfoMessage(string.Empty);
             _interactiveService.LogInfoMessage("Creating Dotnet Publish Zip file...");
 
+            // Since Beanstalk doesn't currently have .NET 7 preinstalled we need to make sure we are doing a self contained publish when creating the deployment bundle.
+            if (recommendation.Recipe.TargetService == RecipeIdentifier.TARGET_SERVICE_ELASTIC_BEANSTALK && recommendation.ProjectDefinition.TargetFramework == "net7.0")
+            {
+                _interactiveService.LogInfoMessage("Using self contained publish since AWS Elastic Beanstalk does not currently have .NET 7 preinstalled");
+                recommendation.DeploymentBundle.DotnetPublishSelfContainedBuild = true;
```

Comment:
I've gotten in the habit of setting the option setting, which now writes-through to the `DeploymentBundle`.

Not aware if any specific issues right now by not doing so, just wanting to avoid future/downstream issues if someone assumes they are in sync.

Reply:
I going to skip this comment because I don't want to have to make the change if injecting the `IOptionSettingHandler` into this code on updating the affecting tests.

Reply:
Ah, didn't check to see if it was there. Okay.